lib/wasm/WasmFinalizeExportsPlugin.js

```diff
@@ -0,0 +1,57 @@
+/*
+	MIT License http://www.opensource.org/licenses/mit-license.php
+	Author Tobias Koppers @sokra
+	*/
+"use strict";
+
+const Queue = require("../util/Queue");
+const WebAssemblyImportDependency = require("../dependencies/WebAssemblyImportDependency");
+const UnsupportedWebAssemblyFeatureError = require("../wasm/UnsupportedWebAssemblyFeatureError");
+
+class WasmFinalizeExportsPlugin {
+	apply(compiler) {
+		compiler.hooks.compilation.tap("WasmFinalizeExportsPlugin", compilation => {
+			compilation.hooks.finishModules.tap(
+				"WasmFinalizeExportsPlugin",
+				modules => {
+					const queue = new Queue();
+
+					let module;
+					let jsIncompatibleExports = [];
+
+					for (const module of modules) {
+						if (module.buildMeta.jsIncompatibleExports) {
+							jsIncompatibleExports.push(
+								...module.buildMeta.jsIncompatibleExports
+							);
+						}
+
+						queue.enqueue(module);
+					}
+
+					while (queue.length > 0) {
+						module = queue.dequeue();
+
+						// 1. if a non WebAssembly module
+						if (module.type.startsWith("webassembly") === false) {
```

Comment:
Try to traverse the graph in reverse.

* For each wasm module
* For each ModuleReasons in `module.reasons`
* `reason.dependency.getReference()` give info about export names if any